pandas/io/json/_json.py

```diff
@@ -934,11 +929,13 @@ def _try_convert_data(self, name, data, use_dtypes=True, convert_dates=True):
             # coerce floats to 64
             try:
                 data = data.astype("int64")
-                result = True
             except (TypeError, ValueError):
                 pass
 
-        return data, result
+        if name == "index" and self.orient == "split" and len(data):
```

Comment:
ok looks a lot better. can you add a comment here and structure this more like

```
# if we have an index, we want to preserve dtypes
if name == "index" and len(data):
    if self.orient == "split":
        return data, False

return data, True
```

Reply:
Done